internal/testutils/specialimage/twoplatform.go

```diff
@@ -14,72 +14,66 @@ import (
 func TwoPlatform(dir string) (*ocispec.Index, error) {
 	const imageRef = "twoplatform:latest"
 
-	layer1Desc, err := writeLayerWithOneFile(dir, "bash", []byte("layer1"))
+	manifest1Desc, err := oneLayerPlatformManifest(dir, platforms.MustParse("linux/amd64"), FileInLayer{Path: "bash", Content: []byte("layer1")})
 	if err != nil {
 		return nil, err
 	}
-	layer2Desc, err := writeLayerWithOneFile(dir, "bash", []byte("layer2"))
+
+	manifest2Desc, err := oneLayerPlatformManifest(dir, platforms.MustParse("linux/arm64"), FileInLayer{Path: "bash", Content: []byte("layer2")})
 	if err != nil {
 		return nil, err
 	}
 
-	config1Desc, err := writeJsonBlob(dir, ocispec.MediaTypeImageConfig, ocispec.Image{
-		Platform: platforms.MustParse("linux/amd64"),
-		Config: ocispec.ImageConfig{
-			Env: []string{"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"},
-		},
-		RootFS: ocispec.RootFS{
-			Type:    "layers",
-			DiffIDs: []digest.Digest{layer1Desc.Digest},
-		},
-	})
+	index := ocispec.Index{
+		Versioned: specs.Versioned{SchemaVersion: 2},
+		MediaType: ocispec.MediaTypeImageIndex,
+		Manifests: []ocispec.Descriptor{manifest1Desc, manifest2Desc},
+	}
+
+	ref, err := reference.ParseNormalizedNamed(imageRef)
 	if err != nil {
 		return nil, err
 	}
+	return multiPlatformImage(dir, ref, index)
+}
 
-	manifest1Desc, err := writeJsonBlob(dir, ocispec.MediaTypeImageManifest, ocispec.Manifest{
-		MediaType: ocispec.MediaTypeImageManifest,
-		Config:    config1Desc,
-		Layers:    []ocispec.Descriptor{layer1Desc},
-	})
+type FileInLayer struct {
+	Path    string
+	Content []byte
+}
+
+func oneLayerPlatformManifest(dir string, platform platforms.Platform, f FileInLayer) (ocispec.Descriptor, error) {
+	layerDesc, err := writeLayerWithOneFile(dir, f.Path, f.Content)
 	if err != nil {
-		return nil, err
+		return ocispec.Descriptor{}, err
 	}
 
-	config2Desc, err := writeJsonBlob(dir, ocispec.MediaTypeImageConfig, ocispec.Image{
-		Platform: platforms.MustParse("linux/arm64"),
+	configDesc, err := writeJsonBlob(dir, ocispec.MediaTypeImageConfig, ocispec.Image{
+		Platform: platform,
 		Config: ocispec.ImageConfig{
 			Env: []string{"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"},
 		},
 		RootFS: ocispec.RootFS{
 			Type:    "layers",
-			DiffIDs: []digest.Digest{layer1Desc.Digest},
+			DiffIDs: []digest.Digest{layerDesc.Digest},
 		},
 	})
 	if err != nil {
-		return nil, err
+		return ocispec.Descriptor{}, err
 	}
 
-	manifest2Desc, err := writeJsonBlob(dir, ocispec.MediaTypeImageManifest, ocispec.Manifest{
+	manifestDesc, err := writeJsonBlob(dir, ocispec.MediaTypeImageManifest, ocispec.Manifest{
 		MediaType: ocispec.MediaTypeImageManifest,
-		Config:    config2Desc,
-		Layers:    []ocispec.Descriptor{layer2Desc},
+		Config:    configDesc,
+		Layers:    []ocispec.Descriptor{layerDesc},
 	})
 	if err != nil {
-		return nil, err
+		return ocispec.Descriptor{}, err
 	}
+	manifestDesc.Platform = &platform
```

Comment:
The descriptor is created by `writeJsonBlob` which doesn't set platform as it's a generic function that just writes a json to the content store.
We patch the platform because the returned manifest will be put in the OCI index - and they are expected to have the platform set, like:

```
{
  "manifests": [
    {
      "digest": "sha256:5f2875f43c30349791f096598cbc96a774c422eeb2405e09ac4a7fc97492b623",
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      },
      "size": 529
    },
    {
      "digest": "sha256:2f17ea683dcddae619a6e443ae40a904f20dc5bd7c06b275dd6b2bab9481be5d",
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "platform": {
        "architecture": "arm",
        "os": "linux",
        "variant": "v5"
      },
      "size": 529
    },

...
```